VideoApp/VideoApp/Stores/Auth/Core/AuthError.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright (C) 2020 Twilio, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+enum AuthError: Error {
+    case expiredPasscode
+    case wrongPasscode
+    case userDisabled
+    case invalidEmail
+    case wrongPassword
+    case networkError
+    case unknown
+    
+    init(apiError: APIError) {
```

Comment:
Those 3 other errors come from Firebase so that is why they are not in APIError. For example neither community API or main API returns a `userDisabled` error. AuthError is a superset of APIError and Firebase errors so that the UI has single error type to deal with for auth problems.